src/devtoolsPanel.ts

```diff
@@ -420,19 +422,20 @@ export class DevToolsPanel {
                 ">
             </head>
             <body>
-                <iframe id="devtools-frame" frameBorder="0" src="${cdnBaseUrl}?experiments=true&theme=${theme}"></iframe>
+                <iframe id="devtools-frame" frameBorder="0" src="${cdnBaseUrl}?experiments=true&theme=${theme}&headless=${this.isHeadless}"></iframe>
             </body>
             </html>
             `;
     }
 
-    private setBrowserRevision(revision: string) {
-        if (revision !== '') {
+    private setBrowserRevision(msg: {revision: string, isHeadless: boolean}) {
+        if (msg.revision !== '') {
             this.config.isCdnHostedTools = true;
-            this.config.devtoolsBaseUri = this.config.devtoolsBaseUri || `https://devtools.azureedge.net/serve_file/${revision}/vscode_app.html`;
+            this.config.devtoolsBaseUri = this.config.devtoolsBaseUri || `https://devtools.azureedge.net/serve_file/${msg.revision}/vscode_app.html`;
```

Comment:
NIT: If the `devtoolsBaseUri` is set in the config, can we check for it and avoid setting up the `BrowserVersionDetectionSocket` / the version detection logic altogether?